### PR TITLE
[FEAT] Sign expression implemtation

### LINKED ISSUE #1917: [EXPRESSIONS] `.sign`
`.sign()`

Return the sign of the input.

### PR DESCRIPTION
**Pull request by sherlockbeard**
implementation for sign resolves #1917

### CODE REVIEW

**Comment by samster25** [comment]
Thanks @sherlockbeard for taking this on! 
@colin-ho Can you take a look?

File: daft/expressions/expressions.py [hunk 1]
```diff
@@ -342,6 +342,11 @@ def floor(self) -> Expression:
         expr = self._expr.floor()
         return Expression._from_pyexpr(expr)
 
+    def sign(self) -> Expression:
+        """The floor of a numeric expression (``expr.sign()``)"""
```
Comment (colin-ho) [inline daft/expressions/expressions.py:346]:
```suggestion
        """The sign of a numeric expression (``expr.sign()``)"""
```

File: src/daft-core/src/series/ops/sign.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+use crate::series::Series;
+use common_error::DaftError;
+use common_error::DaftResult;
+
+impl Series {
+    pub fn sign(&self) -> DaftResult<Series> {
+        use crate::series::array_impl::IntoSeries;
+        use DataType::*;
+        match self.data_type() {
+            UInt8 | UInt16 | UInt32 | UInt64 => Ok(self.clone()),
```
Comment (colin-ho) [inline src/daft-core/src/series/ops/sign.rs:11]: Need to call `.sign()?` on these as well otherwise the value is returned as is.

**Comment by colin-ho** [comment]
> Thanks @sherlockbeard for taking this on! @colin-ho Can you take a look?

Thanks @sherlockbeard for this PR! Just left a few comments :)

File: src/daft-core/src/series/ops/sign.rs (hunk 2 above)
Comment (sherlockbeard) [reply src/daft-core/src/series/ops/sign.rs:11]: is there any way i can trigger this condition from python test ?

Comment (colin-ho) [reply src/daft-core/src/series/ops/sign.rs:11]: Absolutely! You can create a MicroPartition from a pyarrow table with uint columns. For example:
```
table = MicroPartition.from_arrow(
    pa.Table(
        {
            "uint8": pa.array([None, 0, 1, 2, 3], type=pa.uint8()),
        }
    )
)
```

Comment (sherlockbeard) [reply src/daft-core/src/series/ops/sign.rs]: is there no way to 
```impl<T: DaftNumericType> DataArray<T>
where T::Native: Unsigned
{
    pub fn sign(&self) -> DaftResult<Self> {
        self.apply(|v| if v as u64  == 0 { 0 } else { 1 }
)
    }
}
```

currently i have implemented it individually

Comment (colin-ho) [reply src/daft-core/src/series/ops/sign.rs]: You can use the zero and one identities from numtraits https://docs.rs/num-traits/0.2.18/num_traits/identities/index.html
```
impl<T: DaftNumericType> DataArray<T>
where
    T::Native: Unsigned,
{
    pub fn sign_unsigned(&self) -> DaftResult<Self> {
        self.apply(|v| {
            if v.is_zero() {
                T::Native::zero()
            } else {
                T::Native::one()
            }
        })
    }
}
```

Comment (sherlockbeard) [reply src/daft-core/src/series/ops/sign.rs]: ok type casting is hard...
also if i need to do something like v * 2 .
then do i need to impl Int8Array then do operation or i can do it in DaftNumericType

Comment (colin-ho) [reply src/daft-core/src/series/ops/sign.rs]: Indeed it is haha, for arithmetic operations we use a different trait bound, you can check it out here: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/arithmetic.rs#L109

**Review by colin-ho — APPROVED** [review]

**Comment by colin-ho** [comment]
Awesome stuff @sherlockbeard!! Thanks for this contribution :)

**Comment by jaychia** [comment]
Congrats on your first PR @sherlockbeard !! 🔥

**Pull request merged**